### PR TITLE
Fix root array not works with webpack resolve

### PR DESCRIPTION
**Pull request by chemzqm**
Fix of #883, also add support for fallback field of webpack resolve.

http://webpack.github.io/docs/resolving.html

### CODE REVIEW

File: plugin/webpack.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ var tern = require("../lib/tern");
 require("./commonjs");
 require("./es_modules");
 
+Array.isArray = Array.isArray || function(v) {
```
Comment (marijnh) [inline plugin/webpack.js]: Please don't patch global objects from the library. We can use a local `isArray` variable for this.

Comment (LubergAlexander) [inline plugin/webpack.js]: I would like to get this merged, how can I help ? @chemzqm would it be possible for you to fix the comment by @marijnh ? I can do that, but I will probably have to copy your changes to my own branch and issue another pull request

Comment (chemzqm) [inline plugin/webpack.js]: I've fixed that problem, I think it's ready get merged.

Comment (LubergAlexander) [inline plugin/webpack.js]: @chemzqm thanks! that was fast! @marijnh is it possible to merge it ?

**Comment by marijnh** [comment]
Thanks. Merged as db13aac

**Pull request closed without merging**